go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -182,12 +185,15 @@ func (h *secondaryLayer3NetworkControllerEventHandler) UpdateResource(oldObj, ne
 					nodeSubnetChanged ||
 					hostCIDRsChanged(oldNode, newNode) ||
 					nodeGatewayMTUSupportChanged(oldNode, newNode)
+				_, failed = h.oc.syncRerouteFailed.Load(newNode.Name)
+				syncReroute := failed || util.NodeHostCIDRsAnnotationChanged(oldNode, newNode)
```

Comment:
this remote->local zone change logic you had for DNC is not needed here?

Reply:
yeah that was intentional. I didn't think we support (or want to) migrating from non-IC to IC with UDNs enabled.